### PR TITLE
Support visual mark movements

### LINKED ISSUE #2690: Support for `'>` and ``>` to go to end of last selected region
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

Feature request.

**What happened**:

I pressed `'>` or `BACKTICK>` in Visual Studio Code with the VsCodeVim plugin installed after I've previously selected something in visual mode.

**What did you expect to happen**:

I expected to be moved to the end of the last visual region that I selected.

**How to reproduce it (as minimally and precisely as possible)**:

* Select something in visual mode.
* Exit visual mode (e.g. press escape)
* Move around a bit
* Press `'>` or `BACKTICK>`
* Nothing happens

**Background**:

I'm really missing the `'>` (and it's associated `BACKTICK>`) command in VSCodeVim.

This is from the vim manual (`:h '>`):
```
							*'>* *`>*
'>  `>			To the last line or character of the last selected
			Visual area in the current buffer.  For block mode it
			may also be the first character of the last line (to
			be able to define the block).  Note that 'selection'
			applies, the position may be just after the Visual
			area.
```

For example, if you have this code in the editor:

```rust
fn some_func() {
    42
}
```

then you can type `vipy'>p` (view-in-paragraph, yank-visual-area, go-to-end-of-last-visual-area, paste) to make a copy of it:

```rust
fn some_func() {
    42
}
fn some_func() {
    42
}
```

**Environment**:

- Extension (VsCodeVim) version: v0.12.0
- VSCode version: April 2018 (version 1.23)
- OS version: MacOS High Sierra

### PR DESCRIPTION
**Pull request by sql-koala**
**What this PR does / why we need it**:
This adds support for the visual mark motions: '<, '>. Character variants, too.

**Which issue(s) this PR fixes**
Fixes #2690
related to #2349
But does not fix it completely, as the issue is also about ']

**Special notes for your reviewer**:
Not much, we "calculate" the new position to jump to from vimState.lastVisualSelection on the fly.
From my perspective, this is good to go.

### CODE REVIEW

File: src/actions/motion.ts [hunk 1]
```diff
@@ -447,6 +447,73 @@ class CommandPreviousSearchMatch extends BaseMovement {
...
+  private startOrEnd(lastVisualSelection): Position {
+    // marks from vimstate are sorted by direction of selection (moving forward vs backwards).
+    // must sort to document order
+    let [start, end] = sorted(lastVisualSelection.start, lastVisualSelection.end);
+    return this.mark === VisualMark.SelectionStart ? start : end;
+  }
+
+  private inLineCorrection(position: Position): Position {
+    // for ' mark, we must go to BOL.
+    // for `> mark, we must correct by one char left
```
Comment (sql-koala) [inline src/actions/motion.ts:469]: I don't like this nested ?, but that is what the linter does.

File: src/actions/motion.ts [hunk 2]
```diff
@@ -447,6 +447,73 @@ class CommandPreviousSearchMatch extends BaseMovement {
...
+    return this.registerMode === RegisterMode.LineWise
+      ? position.getLineBeginRespectingIndent()
+      : this.mark === VisualMark.SelectionStart
+      ? position
+      : position.getLeft();
+  }
```
Comment (sql-koala) [inline src/actions/motion.ts:475]: For end of visual selection, we have to correct one char to the left.
Is this linked to
https://github.com/VSCodeVim/Vim/pull/5049#issue-452966423 ?

**Comment by sql-koala** [comment]
Btw, this is also affected by this: (but the existing motions, too).
https://github.com/VSCodeVim/Vim/issues/5057

File: src/actions/operator.ts [hunk 3]
```diff
@@ -304,11 +304,6 @@ export class YankOperator extends BaseOperator {
 
     Register.put(text, vimState, this.multicursorIndex);
 
-    if (vimState.currentMode === Mode.Visual || vimState.currentMode === Mode.VisualLine) {
-      vimState.historyTracker.addMark(start, '<');
-      vimState.historyTracker.addMark(end, '>');
-    }
-
     await vimState.setCurrentMode(Mode.Normal);
```
Comment (sql-koala) [inline src/actions/operator.ts:307]: This is probably old code. lastVisualSelection is tracked in vimState and motions must access this, we dont need/sould delete this.

File: src/actions/motion.ts (hunk 2 above)
Comment (J-Fields) [reply src/actions/motion.ts:564]: Yeah I think so. It's fine for now

**Comment by J-Fields** [comment]
The column doesn't seem to be right for visual line mode, but this looks good other than that. Thanks @sql-koala!

**Comment by sql-koala** [comment]
Yay, merged 👍

**Pull request merged**